app/questionnaire/placeholder_transforms.py

```diff
@@ -80,14 +81,24 @@ def format_number(self, number):
         return ""
 
     @staticmethod
-    def calculate_years_difference(first_date, second_date):
-        return str(
-            relativedelta(
-                PlaceholderTransforms.parse_date(second_date),
-                PlaceholderTransforms.parse_date(first_date),
-            ).years
+    def calculate_difference(first_date, second_date):
+        time = relativedelta(
+            PlaceholderTransforms.parse_date(second_date),
+            PlaceholderTransforms.parse_date(first_date),
         )
 
+        if time.years:
+            if time.years == 1:
+                return f'{time.years} {lazy_gettext("year")}'
+            return f'{time.years} {lazy_gettext("years")}'
+        if time.months:
+            if time.months == 1:
+                return f'{time.months} {lazy_gettext("month")}'
+            return f'{time.months} {lazy_gettext("months")}'
+        if time.days == 1:
+            return f'{time.days} {lazy_gettext("day")}'
+        return f'{time.days} {lazy_gettext("days")}'
```

Comment:
Babel has support for this with `format_timedelta` 
 - http://babel.pocoo.org/en/latest/api/dates.html#babel.dates.format_timedelta. We should look to use that unless it's unsuitable.

Reply:
I used format_timedelta and it's 3 lines now. Thanks!